compiler/vm/vm.nim

```diff
@@ -271,67 +349,124 @@ proc asgnComplex(x: var TFullReg, y: TFullReg) =
   x.ensureKind(y.kind)
   case x.kind
   of rkNone: discard
-  of rkInt: x.intVal = y.intVal
-  of rkFloat: x.floatVal = y.floatVal
-  of rkNode: x.node = copyValue(y.node)
+  #of rkInt: x.intVal = y.intVal
+  #of rkFloat: x.floatVal = y.floatVal
+  of rkAtom: x.atomVal = y.atomVal
   of rkRegisterAddr: x.regAddr = y.regAddr
-  of rkNodeAddr: x.nodeAddr = y.nodeAddr
+  #of rkNodeAddr: x.nodeAddr = y.nodeAddr
 
-proc fastAsgnComplex(x: var TFullReg, y: TFullReg) =
+proc moveComplex(x : var TFullReg, y : var TFullReg) =
+  ## Assign `x` to `y` while moving from `y`. `y` is left in an empty state
   x.ensureKind(y.kind)
   case x.kind
   of rkNone: discard
-  of rkInt: x.intVal = y.intVal
-  of rkFloat: x.floatVal = y.floatVal
-  of rkNode: x.node = y.node
+  #of rkInt: x.intVal = move y.intVal
+  #of rkFloat: x.floatVal = move y.floatVal
+  of rkAtom: x.atomVal = move y.atomVal
+  of rkRegisterAddr: x.regAddr = move y.regAddr
+  #of rkNodeAddr: x.nodeAddr = move y.nodeAddr
+
+  # Explicitly reset y to empty
+  y.ensureKind(rkNone)
+
+proc fastAsgnComplex(x: var TFullReg, y: TFullReg) {.deprecated: "Same as `asgnComplex`".} =
+  x.ensureKind(y.kind)
+  case x.kind
+  of rkNone: discard
+  #of rkInt: x.intVal = y.intVal
+  #of rkFloat: x.floatVal = y.floatVal
+  of rkAtom: x.atomVal = y.atomVal
   of rkRegisterAddr: x.regAddr = y.regAddr
-  of rkNodeAddr: x.nodeAddr = y.nodeAddr
+  #of rkNodeAddr: x.nodeAddr = y.nodeAddr
+
+# TODO: rename (any proposal?)
+func writeAtom(n: var TNode, x: VmObjectAtom, setupNode : static[bool]) = 
```

Comment:
Need to understand the intension.

Is it meant to write the value of the atom into the nimnode or load the value from the node into the atom?

Reply:
Basically to turn the atom back into the `PNode` representation that the VM used prior

Reply:
`writeToPNode` or `uodatePNode`